src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1230,6 +1283,47 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
             }
         }
 
+        private static string GetTypeName(
+            Address typeID,
+            Dictionary<Address, string> typeNameCache,
+            Dictionary<Address, GCBulkTypeValues> typeMapping)
+        {
+            string typeName;
+            if (typeNameCache.TryGetValue(typeID, out typeName))
+                return typeName;
+
+            GCBulkTypeValues bulkTypeValues;
+            if (typeMapping.TryGetValue(typeID, out bulkTypeValues))
+            {
+                typeName = bulkTypeValues.TypeName;
+                var parameterCount = bulkTypeValues.TypeParameterCount;
+                if (parameterCount > 0)
```

Comment:
📝 This is currently producing results like this:

```
System.WeakReference`1[System.Diagnostics.Tracing.EtwSession]<System.Diagnostics.Tracing.EtwSession>
```

❓ It seems `GCBulkTypeValues.TypeName` includes generic arguments already, but in a different form. Which of the following would you prefer the result be?

1. ``System.WeakReference`1[System.Diagnostics.Tracing.EtwSession]``
2. `System.WeakReference<System.Diagnostics.Tracing.EtwSession>`

If I go with the first approach, I can change `GCStats.TypeMapping` to just a `Dictionary<Address, string>` and avoid cloning the trace event. But the performance overhead didn't seem to necessitate one way or the other.